examples/framesjs-starter/app/examples/new-api-mint-button/frames/route.tsx

```diff
@@ -38,37 +35,45 @@ const nfts: {
 
 export const frames = createFrames({
   basePath: "/examples/new-api-mint-button/frames",
-  initialState: {
-    pageIndex: 0,
-  },
 });
 
-const handleRequest = frames(async ({ pressedButton, message, state }) => {
-  return {
-    image: nfts[state.pageIndex]!.src,
-    imageOptions: {
-      aspectRatio: "1:1",
-    },
-    buttons: [
-      <Button
-        action="post"
-        state={{ pageIndex: (state.pageIndex - 1) % nfts.length }}
-      >
-        ←
-      </Button>,
-      <Button
-        action="post"
-        state={{ pageIndex: (state.pageIndex + 1) % nfts.length }}
-      >
-        →
-      </Button>,
-      <Button action="mint" target={nfts[state.pageIndex]!.tokenUrl}>
-        Mint
-      </Button>,
-    ],
-    textInput: "Type something!",
-  };
-});
+const handleRequest = frames(
+  async ({ pressedButton, message, searchParams }) => {
+    const page = Number(searchParams?.pageIndex ?? 0);
```

Comment:
ah I see, please when you resolve the conflicts, accept only your changes, only relevant thing from my side was not destructuring the `ctx`

Reply:
not sure I follow, do you want it to be `async (ctx) => {`?